contract-message-test/tests/harness.rs

```diff
@@ -1,184 +1,672 @@
+pub const RANDOM_SALT2: &str = "0xd5f55c10bc830755278e6d2b9278b4177b8bca401a896eb0d3e7710eee293786";
+
+// Test that input messages can be relayed to a contract
+// and that the contract can successfully parse the message data
+mod success {
+    use std::str::FromStr;
+
+    use crate::utils::environment as env;
+    use fuels::prelude::Contract;
+    use fuels::prelude::Salt;
+    use fuels::prelude::StorageConfiguration;
+    use fuels::prelude::TxParameters;
+    use fuels::test_helpers::DEFAULT_COIN_AMOUNT;
+    use fuels::tx::{
+        Address, AssetId, Bytes32, ContractId, Input, Output, TxPointer, UtxoId, Word,
+    };
+
+    pub const RANDOM_SALT: &str =
+        "0x1a896ebd5f55c10bc830755278e6d2b9278b4177b8bca400d3e7710eee293786";
+    pub const RANDOM_SALT2: &str =
+        "0xd5f55c10bc830755278e6d2b9278b4177b8bca401a896eb0d3e7710eee293786";
+
+    #[tokio::test]
+    async fn relay_message_with_predicate_and_script_constraint() {
+        let data_word = 54321u64;
+        let data_bytes = Bytes32::from_str(RANDOM_SALT).unwrap();
+        let data_address = Address::from_str(RANDOM_SALT2).unwrap();
+        let mut message_data = data_word.to_be_bytes().to_vec();
+        message_data.append(&mut env::decode_hex(RANDOM_SALT));
+        message_data.append(&mut env::decode_hex(RANDOM_SALT2));
+        let message_data = env::prefix_contract_id(message_data).await;
+        let message = (100, message_data);
+        let coin = (DEFAULT_COIN_AMOUNT, AssetId::default());
+
+        // Set up the environment
+        let (wallet, test_contract, contract_input, coin_inputs, message_inputs) =
+            env::setup_environment(vec![coin], vec![message]).await;
+
+        // Relay the test message to the test contract
+        let _receipts = env::relay_message_to_contract(
+            &wallet,
+            message_inputs[0].clone(),
+            contract_input,
+            &coin_inputs[..],
+            &vec![],
+            &vec![],
+        )
+        .await;
+
+        // Verify test contract received the message
+        let test_contract_counter = test_contract.get_test_counter().call().await.unwrap().value;
+        assert_eq!(test_contract_counter, 1);
+
+        // Verify test contract received the correct data
+        let test_contract_id: ContractId = test_contract._get_contract_id().into();
+        let test_contract_data1 = test_contract.get_test_data1().call().await.unwrap().value;
+        assert_eq!(test_contract_data1, test_contract_id);
+        let test_contract_data2 = test_contract.get_test_data2().call().await.unwrap().value;
+        assert_eq!(test_contract_data2, data_word);
+        let test_contract_data3 = test_contract.get_test_data3().call().await.unwrap().value;
+        assert_eq!(test_contract_data3, data_bytes.to_vec()[..]);
+        let test_contract_data4 = test_contract.get_test_data4().call().await.unwrap().value;
```

Comment:
```suggestion
        let test_contract_data2 = test_contract.get_test_data2().call().await.unwrap().value;
        let test_contract_data3 = test_contract.get_test_data3().call().await.unwrap().value;
        let test_contract_data4 = test_contract.get_test_data4().call().await.unwrap().value;

        assert_eq!(test_contract_data1, test_contract_id);
        assert_eq!(test_contract_data2, data_word);
        assert_eq!(test_contract_data3, data_bytes.to_vec()[..]);
```

It's a good practice to separate your setup from the tests themselves so that the reader does not have to carefully inspect each line of your test in order to find something. This way they can conceptualize the "block" for setup and then the tests. If they are in the same order as well then they can associate the first setup line with the first assertion etc. This further reduces cognitive load when reviewing. Something I would also consider is combining this section of tests with the counter above.